### PR TITLE
[TH2-1416] "TableSearch" action implemented. Command line arguments in "Open" action added.

### PR DESCRIPTION
**Pull request by MrJonniK**

### CODE REVIEW

File: src/main/java/com/exactprosystems/remotehand/windows/actions/Open.java [hunk 1]
```diff
@@ -54,13 +55,17 @@ public String run(WindowsDriverWrapper driverWrapper, Map<String, String> params
...
+		String appArgs = params.get("appargs");
 
 		Path workDirPath = Paths.get(workDir).normalize();
 		Path exeFilePath = workDirPath.resolve(execFile).normalize();
 
 		DesiredCapabilities capabilities = driverWrapper.createCommonCapabilities();
 		capabilities.setCapability(MobileCapabilityType.APP, exeFilePath.toString().replace('/', '\\'));
 		capabilities.setCapability("appWorkingDir", workDirPath.toString().replace('/', '\\'));
+		if (StringUtils.isNotBlank(appArgs)) {
+			capabilities.setCapability("appArguments", appArgs);
```
Comment (Vladimir-Panarin-Exactpro) [inline src/main/java/com/exactprosystems/remotehand/windows/actions/Open.java:67]: Are these changes expected in this commit? Looks like this is from different task.

File: src/main/java/com/exactprosystems/remotehand/windows/actions/TableSearch.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+		} else {
+			return row.findElement(columnLocator);
+		}
+	}
+
+	private void setTimeOut(WindowsDriverWrapper driverWrapper, int seconds) throws ScriptExecuteException {
+		driverWrapper.getDriver().manage().timeouts().implicitlyWait(seconds, TimeUnit.SECONDS);
+	}
+
+	private static Map<String, String> getFilters(String filters) {
```
Comment (Vladimir-Panarin-Exactpro) [inline src/main/java/com/exactprosystems/remotehand/windows/actions/TableSearch.java]: Move it to separate utilities class, maybe?

File: src/main/java/com/exactprosystems/remotehand/windows/actions/Open.java (hunk 1 above)
Comment (MrJonniK) [reply src/main/java/com/exactprosystems/remotehand/windows/actions/Open.java:67]: I added this change to original task as requirements.

**Review by Vladimir-Panarin-Exactpro — DISMISSED** [review]
@battlemaster Please check as well.

File: src/main/java/com/exactprosystems/remotehand/windows/actions/TableSearch.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+	}
+	
+	private void handleInputParams(Map<String, String> params) throws ScriptExecuteException
+	{
+		if (getId() == null) {
+			throw new ScriptExecuteException("Id is not specified");
+		}
+		firstRowIndex = Integer.parseInt(params.getOrDefault(FIRST_ROW_INDEX, DEFAULT_FIRST_ROW_NUMBER));
+		columnIndex = Integer.parseInt(params.getOrDefault(INDEX, "0"));
+		rowNameFormat = params.getOrDefault(ROW_NAME_FORMAT, DEFAULT_ROW_NAME_FORMAT);
```
Comment (std000) [inline src/main/java/com/exactprosystems/remotehand/windows/actions/TableSearch.java:106]: This parameter is unreachable because all keys is in lowercase

File: src/main/java/com/exactprosystems/remotehand/windows/actions/TableSearch.java [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+		try {
+			handleInputParams(params);
+			ElementSearcher elementSearcher = new ElementSearcher(params, driverWrapper.getDriver(), cachedElements);
+			WebElement table = elementSearcher.searchElement();
+			setTimeOut(driverWrapper, 0);
+
+			WebElement row;
+			boolean rowFound = false;
+			int i = firstRowIndex;
+			while (!rowFound) {
```
Comment (std000) [inline src/main/java/com/exactprosystems/remotehand/windows/actions/TableSearch.java:73]: Deprecated TableClick action allows us to use different indexes with RowName and child ValueName.
So there are Row started with 1, but values started with 0
But there is no possible to do this.

File: src/main/java/com/exactprosystems/remotehand/windows/actions/ClickContextMenu.java [hunk 5]
```diff
@@ -40,8 +40,10 @@ public String run(WindowsDriverWrapper driverWrapper, Map<String, String> params
 		
 		ElementSearcher es = new ElementSearcher(params, driver, cachedWebElements);
 		WebElement element = es.searchElement();
-		
-		element.click();
+
```
Comment (std000) [inline src/main/java/com/exactprosystems/remotehand/windows/actions/ClickContextMenu.java:43]: What is the reason of this changes?

**Review by std000 — CHANGES_REQUESTED** [review]

File: src/main/java/com/exactprosystems/remotehand/windows/actions/ClickContextMenu.java (hunk 5 above)
Comment (MrJonniK) [reply src/main/java/com/exactprosystems/remotehand/windows/actions/ClickContextMenu.java:43]: In some cases when you execute "element.click()" on menu items, an "An element command could not be completed because the element is not pointer- or keyboard interactable" error occurs. This change fixes this issue.

**Review by std000 — APPROVED** [review]
